clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -780,6 +784,15 @@ std::string PointerVariableConstraint::mkString(Constraints &CS,
     EmittedName = true;
   uint32_t TypeIdx = 0;
 
+  // If we're set a GenericIndex for void, it means we're converting it into
+  // a generic function so give it the default generic type name.
```

Comment:
Are sure that it will be generic at this point, or is it just potentially a generic? I haven't looked at the whole codebase, so I'm not sure what constraints might have affected the code determination to this point.

Reply:
`NewGenericIndex` marks it as generic after analysis, or in the original. `BaseType` would be "T" if it was in the original.
The potential generic check comes before changing `NewGenericIndex`.